Services/DataBaseService/MigrationEngine.cs

```diff
@@ -0,0 +1,225 @@
+                        alreadyExecuted = false;
+                    }
+
+                    if (!alreadyExecuted)
+                        scriptsToExecute.Add(fileName);
+                }
+            }
+            return scriptsToExecute;
+        }
+
+        private ExecutedScript CreateDatabase()
+        {
+            var connectionString = configuration.GetConnectionString("MigrationString");
+            var createDbScript = "IF DB_ID('TradingStation') IS NULL CREATE DATABASE[TradingStation];";
+
+            using (var conn = new SqlConnection(connectionString))
+            {
+                conn.Open();
+                try
+                {
+                    using (SqlCommand command = new SqlCommand(createDbScript, conn))
+                        command.ExecuteNonQuery();
+                    // TODO replace with logs
+                    Console.WriteLine("\tTradingStation DB was created successfully or already existed.");
+                    return new ExecutedScript(DateTime.Now, createDbScript);
+                }
+                catch (SqlException e)
+                {
+                    // TODO replace with logs
+                    Console.WriteLine(e.Message + "\n\tCouldn't create the TradingStation DB.");
+                    throw;
+                }
+            }
+        }
+
+        private ExecutedScript CreateScriptsTable(string connectionString)
+        {
+            var createTableScript = new StringBuilder();
+            createTableScript.Append("IF OBJECT_ID('dbo.ExecutedScripts', 'U') IS NULL ");
+            createTableScript.Append("CREATE TABLE [dbo].[ExecutedScripts] "); 
+            createTableScript.Append("(Id INT IDENTITY NOT NULL, ");
+            createTableScript.Append("FileName NVARCHAR(100) NOT NULL, ");
+            createTableScript.Append("ExecutionTime DATETIME NOT NULL, ");
+            createTableScript.Append("Code NVARCHAR(MAX) NOT NULL, ");
+            createTableScript.Append("CONSTRAINT PKscripts PRIMARY KEY CLUSTERED(Id));");
+
+            using (var conn = new SqlConnection(connectionString))
+            {
+                conn.Open();
+                try
+                {
+                    using (SqlCommand command = new SqlCommand(createTableScript.ToString(), conn))
+                        command.ExecuteNonQuery();
+                    // TODO replace with logs
+                    Console.WriteLine("\tThe Scripts table was created successfully or already existed.");
+                    return new ExecutedScript(DateTime.Now, createTableScript.ToString());
+                }
+                catch (SqlException e)
+                {
+                    // TODO replace with logs
+                    Console.WriteLine(e.Message + "\n\tCouldn't create the Scripts table.");
+                    throw;
+                }
+            }
+        }
+
+        internal struct ExecutedScript
```

Comment:
move this in another file

Reply:
Why struct not class ?

Reply:
Originally -- since it's immutable and locally-used. However, I agree that the Code property may be a lot larger than 16bit, so now it's a class